engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -70,6 +86,32 @@ internal object NadelHydrationFieldsBuilder {
         }
     }
 
+    private fun filterObjectTypeNamesToMatchActorField(
+        field: ExecutableNormalizedField,
+        actorFieldOverallObjectTypeNames: Set<String>
+    ) {
+        field.objectTypeNames.removeIf { !actorFieldOverallObjectTypeNames.contains(it) }
+    }
+
+    private fun getActorFieldOverallObjectTypenames(
+        instruction: NadelBatchHydrationFieldInstruction,
+        executionBlueprint: NadelOverallExecutionBlueprint
+    ): Set<String> {
+        val actorFieldUnderlyingType = instruction.actorFieldDef.type.unwrapAll()
+        val overallTypeName =
+            executionBlueprint.getOverallTypeName(instruction.actorService, actorFieldUnderlyingType.name)
+        val actorFieldOverallObjectTypes: List<GraphQLType> =
+            when (val actorFieldOverallType = executionBlueprint.schema.getType(overallTypeName)!!) {
+                is GraphQLInterfaceType -> executionBlueprint.schema.getImplementations(actorFieldOverallType)
+                is GraphQLUnionType -> actorFieldOverallType.types
+                else -> {
+                    listOf(actorFieldOverallType)
+                }
+            }
+        return actorFieldOverallObjectTypes.map { (it as GraphQLNamedType).name }
+            .toSet()
+    }
+
```

Comment:
Other parts of Nadel have excellent comments in `graphql psudeo code form` that allow you to explain what we are doding in complicated type filtering methods.  We should do this more often